### ISSUE TITLE
Android app crashes on session disconnected

### ISSUE THREAD

**Issue by selcute** [opentok/opentok-react-native#175]
# Bug Report

**What is the current *bug* behavior?**
App crashes on session disconnected on Android when we force disconnect user from the session and he was also in this session before. It seems like the session isn't closing and doesn't clear its callbacks. The ```sessionConnected``` event fires every time user joins the session, but ```sessionDisconnected``` fires only when we force close the session/disconnect all users.

My temporary solution is: #176 but it doesn't feel good.

**Steps to reproduce**
~~Set ```sessionDisconnected``` event handler on OTSession component, join video chat and force disconnect users from the session~~
Doesn't matter whether even handler is set.
1. Join an existing video chat session and then leave the session (unmount component) - if ```sessionDisconnected``` event handler is set it doesn't fire.
2. Join again this session and then force disconnect user from this session (somehow in the backend or whatever).
3. App crashes

**What is the expected *correct* behavior?**
App shouldn't crash.

**Relevant logs and/or screenshots**
```
Fatal Exception: java.lang.RuntimeException: Illegal callback invocation from native module. This callback type only permits a single invocation from native code.
       at com.facebook.react.bridge.CallbackImpl.invoke(CallbackImpl.java:28)
       at com.opentokreactnative.OTSessionManager.onDisconnected(OTSessionManager.java:484)
       at com.opentok.android.Session.onDisconnected(Session.java:1605)
       at com.opentok.android.Session$5.run(Session.java:1566)
       at android.os.Handler.handleCallback(Handler.java:790)
       at android.os.Handler.dispatchMessage(Handler.java:99)
       at android.os.Looper.loop(Looper.java:164)
       at android.app.ActivityThread.main(ActivityThread.java:7000)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:441)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1408)
```

**Comment by Artemnsk** [comment]
Maybe something similar **on iOS** but `sessionDisconnected` simply not executed?
That's what I did for testing. In my "root" _App.js_ file I set up native events like this:

```
constructor(props) {
  super(props)
  const sessionEvents = sanitizeSessionEvents({
    sessionDisconnected: (session, a, b) => {
      // Never executed!
    },
    sessionConnected: (session, a, b) => {
      // Executed as expected.
    },
    // ... other event handlers that work fine.
  })
  setNativeEvents(sessionEvents)
}
```

.. And I commented `removeNativeEvents(...)` execution in `componentWillUnmount()` of `OTSession` component. Thus, `OT.disconnectSession()` still run but I do not remove event handlers any more to be 100% sure I executed `sessionDisconnected()` declared in _App.js_.

However `sessionDisconnected` never run. But `sessionConnected()` as well as any other event handlers declared in _App.js_ executed fine.

**Comment by selcute** [comment]
Still happening in v0.9

**Comment by msach22** [comment]
@selcute @Artemnsk I have a fix for the crash and the `sessionDisconnected` event - I will patch this by tomorrow.

**Comment by msach22** [comment]
The fix for this has been released with v0.9.4
